packages/odyssey-react/src/components/TextInput/TextInput.tsx

```diff
@@ -68,14 +69,40 @@ interface CommonProps
   placeholder?: string;
 
   /**
-   * An optional string prefix displayed before the input
+   * An optional prefix button displayed before the input
+   * Only one of PrefixButton, PrefixIcon, PrefixText can be set
    */
-  prefix?: string;
+  PrefixButton?: ReactNode;
```

Comment:
using capitalized props isn't something we've done before - what was the thinking here?

Reply:
It's to signal that the prop is `ReactNode` and not a primitive type.

Reply:
got it thanks for the context, I think lowercase prop names should be fine for that too, especially considering that `ReactNode` does contain some primitive types like number string undefined etc.